### PR TITLE
Move out private resources into a separate private repo

### PR DESCRIPTION
**Pull request by siamalekpour**
HIdding private resources from the public eye.

https://trello.com/c/6KWY9NZj

### CODE REVIEW

**Comment by siamalekpour** [comment]
[Take out private resources from gapi-js and gapi-redux and place in separate project](https://trello.com/c/6KWY9NZj/463-take-out-private-resources-from-gapi-js-and-gapi-redux-and-place-in-separate-project)

File: src/gapi-js.js [hunk 1]
```diff
@@ -2,128 +2,138 @@ import request from 'superagent';
...
-export default class Gapi extends GapiResources {
+export const GapiGenerator = (ResourceClass) => {
 
-  constructor ({url='https://rest.gadventures.com', key, proxy}){
-    super();
+  class Gapi extends ResourceClass {
 
-    if( !key ) {
-      throw 'A gapi key is required when instantiating Gapi'
+    constructor ({url = 'https://rest.gadventures.com', key, proxy}) {
```
Comment (siamalekpour) [inline src/gapi-js.js]: Everything down this line is the same code. Just tabbed. No need for a review

File: src/gapi-js.js [hunk 2]
```diff
@@ -2,128 +2,138 @@ import request from 'superagent';
 import GapiResources from './gapi-resources';
 
 
-export default class Gapi extends GapiResources {
+export const GapiGenerator = (ResourceClass) => {
```
Comment (siamalekpour) [inline src/gapi-js.js]: Concept is simple. Class `Gapi` used to extend from `GapiResources`. Now we'd want it to extend from a dynamic class. Could be `GapiResources` or from our private repo `GapiResourcesPrivate`.

I wrapped the class we already had, with a function that accepts the resource class as an argument and returns a  class that inherits from it.

**Comment by siamalekpour** [comment]
The private repo's at: https://github.com/gadventures/gapi-js-private

File: src/gapi-resources.js [hunk 3]
```diff
@@ -13,24 +13,6 @@ export default class GapiResources {
   get suppliers() { this.resource = 'suppliers'; return this; }
   set suppliers(value) { return this; }
```
Comment (thyleung) [inline src/gapi-resources.js]: Should `suppliers` be removed to the private fields as well?

Comment (siamalekpour) [reply src/gapi-resources.js]: I'll move it out

**Comment by thyleung** [comment]
Just need to move `suppliers` to https://github.com/gadventures/gapi-js-private/blob/master/src/gapi-resources-private.js.  Otherwise, looks good to me 👍

**Comment by siamalekpour** [comment]
@thyleung Moved `suppliers` out

**Comment by thyleung** [comment]
👍

**Pull request merged**